File: docs/src/testers/sessions/lifecycle/stop_session.ts
// @sniptest filename=stop_session.ts
import { NotteClient } from 'notte-sdk';

const client = new NotteClient();

// Manual lifecycle: always stop in finally.
const session = client.Session({ idle_timeout_minutes: 2 });
await session.start();
try {
  const status = await session.status();
  if (status.status !== 'active') throw new Error('Session is not active');
  console.log(
    JSON.stringify({
      session_id: session.getId(),
      status: status.status,
      idle_timeout_minutes: status.idle_timeout_minutes,
    }),
  );
} finally {
  await session.stop();
}

File: docs/src/testers/sessions/lifecycle/stop_session.py
# @sniptest filename=stop_session.py
import json

from notte_sdk import NotteClient

client = NotteClient()

# Manual lifecycle: always stop in finally.
session = client.Session(idle_timeout_minutes=2)
session.start()
try:
    status = session.status()
    if status.status != "active":
        raise RuntimeError("Session is not active")
    print(
        json.dumps(
            {
                "session_id": session.session_id,
                "status": status.status,
                "idle_timeout_minutes": status.idle_timeout_minutes,
            }
        )
    )
finally:
    session.stop()
